waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/DatabaseMappingService.java

```diff
@@ -37,6 +37,14 @@
    */
   DatabaseMapping databaseMapping(@NotNull String databaseName) throws NoSuchObjectException;
 
+  void checkTable(String databaseName, String tableName,
```

Comment:
This could also do with some javadoc explaining what exactly it's checking and in what case the `NoSuchObjectException` will be thrown.